utils/directories.cc

```diff
@@ -140,7 +148,23 @@ future<> directories::do_verify_owner_and_mode(fs::path path, recursive recurse,
 };
 
 future<> directories::verify_owner_and_mode(fs::path path, recursive recursive) {
-    return do_verify_owner_and_mode(std::move(path), recursive, 0);
+    return do_verify_owner_and_mode(std::move(path), recursive, 0, [](fs::path &path){return true;});
+}
+
+future<> directories::verify_owner_and_mode_of_data_dir(directories::set dir_set) {
+    co_await parallel_for_each(dir_set.get_paths(), [](const auto &path) -> future<> {
+        co_await do_verify_owner_and_mode(std::move(path), recursive::yes, 0, [](fs::path &path) {
+            const sstring &file_name = path.filename().string();
+            return file_name.ends_with(sstables::sstable_version_constants::DATA_SUFFIX) || file_name.ends_with(sstables::sstable_version_constants::INDEX_SUFFIX);
+        });
+    });
+
+    co_await parallel_for_each(dir_set.get_paths(), [](const auto &path) -> future<> {
+        co_await do_verify_owner_and_mode(std::move(path), recursive::yes, 0, [](fs::path &path) {
+            const sstring &file_name = path.filename().string();
+            return !(file_name.ends_with(sstables::sstable_version_constants::DATA_SUFFIX) || file_name.ends_with(sstables::sstable_version_constants::INDEX_SUFFIX));
```

Comment:
this block us similar to the above that I wonder if we can encapsulate it in a reusable function

Reply:
I have now updated this code to run in a loop to achieve the same results.